src/java.base/share/classes/java/lang/invoke/TypeConvertingMethodAdapter.java

```diff
@@ -25,34 +25,43 @@
 
 package java.lang.invoke;
 
-import jdk.internal.org.objectweb.asm.MethodVisitor;
-import jdk.internal.org.objectweb.asm.Opcodes;
-import jdk.internal.org.objectweb.asm.Type;
+import java.lang.constant.ClassDesc;
+import java.lang.constant.ConstantDescs;
+import java.lang.constant.MethodTypeDesc;
+import java.util.Map;
+import jdk.internal.classfile.Classfile;
+import jdk.internal.classfile.CodeBuilder;
+import jdk.internal.classfile.Opcode;
+import jdk.internal.classfile.TypeKind;
 import sun.invoke.util.BytecodeDescriptor;
 import sun.invoke.util.Wrapper;
 import static sun.invoke.util.Wrapper.*;
 
-class TypeConvertingMethodAdapter extends MethodVisitor {
-
-    TypeConvertingMethodAdapter(MethodVisitor mv) {
-        super(Opcodes.ASM7, mv);
-    }
+class TypeConvertingMethodAdapter {
```

Comment:
This really is a utility class providing instructions to perform primitive type conversions. If they are valuable, we can probably add them to Classfile API components in a separate patch, too.